core/src/main/java/org/elasticsearch/bootstrap/Security.java

```diff
@@ -212,7 +226,36 @@ static Permissions createPermissions(Environment environment) throws IOException
             // we just need permission to remove the file if its elsewhere.
             policy.add(new FilePermission(environment.pidFile().toString(), "delete"));
         }
-        return policy;
+    }
+    
+    static void addBindPermissions(Permissions policy, Settings settings) throws IOException {
+        // http is simple
+        String httpRange = settings.get("http.netty.port", 
+                               settings.get("http.port", 
+                                       NettyHttpServerTransport.DEFAULT_PORT_RANGE));
+        policy.add(new SocketPermission("localhost:" + httpRange, "listen,resolve"));
+        // transport is waaaay overengineered
+        Map<String, Settings> profiles = settings.getGroups("transport.profiles", true);
```

Comment:
I agree with the all the arguments around the transport profile handling code. What I'm saying is that we should reuse it (not copy it!) in Security to make sure we reach the same conclusions. This will also help prevent bugs when we fix the issues in the NettyTransport's settings code.

Here is what I meant : https://github.com/elastic/elasticsearch/commit/ac516f6b733b8ceddd9029fe87a0e12ceeb2aab1?w=1

IMHO, a good move while not violating @rmuir *valid* requirement not to pull the code into Security.java